src/Akka.Persistence.Sql/Query/QueryThrottler.cs

```diff
@@ -40,31 +59,72 @@ internal sealed class ReturnQueryStart
     private ReturnQueryStart() { }
 }
 
+#region Test classes
+
+/// <summary>
+/// For testing purposes
+/// </summary>
+internal sealed class GetUsedPermits
+{
+    public static readonly GetUsedPermits Instance = new();
+    private GetUsedPermits() { }
+}
+
+/// <summary>
+/// For testing purposes
+/// </summary>
+internal sealed class GetPendingRequests
+{
+    public static readonly GetPendingRequests Instance = new();
+    private GetPendingRequests() { }
+}
+
+internal sealed class GetWatchCount
+{
+    public static readonly GetWatchCount Instance = new();
+    private GetWatchCount() { }
+}
+
+#endregion
+
 /// <summary>
 /// Token bucket throttler that grants queries permissions to run each iteration
 /// </summary>
 /// <remarks>
-/// Works identically to the RecoveryPermitter built into Akka.Persistence.
+/// Works almost identically to the RecoveryPermitter built into Akka.Persistence.
+/// 
+/// NOTE: Since this permitter works with Ask and we can't death watch an Ask temp
```

Comment:
It would be nice if we could encapsulate/sugar this behavior in future, that said comment is great breadcrumb for future improvement.